Fixed #1714 and #1791

linzhp opened the pull request:
Issues and solutions:

#1714: the `cgo_deps` parameter of a `go_library` rule was not copied to the `go_test` rules targeting at the `go_library` rule. So when the `go_test` rules were built, the `cgo_library` rules needed by the `go_library` were not built.

Solution: Copy `cgo_deps` from `go_library` to `go_test`

#1791: When Buck was calling cgo to generate the import file, it sets the package name to be the full package name with paths. As a result, the `package` statement in the resulting Go file has the full package name with `/` and `.` in it, causing syntax error when the Go file is compiled.

Solution: Only use the last part of the package name in the cgo-generated go source

ttsugriy commented on test/com/facebook/buck/go/testdata/go_test/cgo/lib/double_cgo.go:
```diff
@@ -0,0 +1,12 @@
+package lib
+
+/*
```
why is this method commented out? Remove?

linzhp replied:
It is a method implemented in C programming language. For more information about Go calling into code written in C, please refer to the [official cgo document](https://golang.org/cmd/cgo/).

linzhp commented:
No worries. I've updated summary, and picked some changes from #1782 so more tests can run on Travis CI. Please take another look.

linzhp commented:
I tried to do that, but with my PRs not getting reviewed for many days, it's sometimes hard:( 

This PR was created just for fixing #1714 (you could see I changed the title for this PR in the middle). But it was waiting here for a while without any attention. When I was trying to fix #1791, I found I could modify and reuse the test in this PR to cover both scenarios, rather than creating a different set of test/test data with minor differences. So I just changed it on top of this PR.

As an opposite, as I don't want to mix this PR with #1782 (which BTW is getting any more feedback for almost a week), I had to pick changes needed here carefully from that PR

Thanks for approving this PR. Hope it will get merged soon.

ttsugriy commented:
I'm sorry about the delays with reviews @linzhp - unfortunately nobody on Buck's core team is familiar with Go and it was written by external contributors, it's hard to find good reviewers for your changes :(

linzhp commented:
@ttsugriy I saw you were landing this PR a day ago, but it's not merged yet. Is it stuck in somewhere in the landing process?

ttsugriy commented:
@linzhp, I'm sorry for the delay - we had issues with our internal CI, but I'm working on landing it. I'm committed to landing in no more than 2 hours.
